python/variable.cpp

```diff
@@ -878,10 +879,84 @@ void init_variable(py::module &m) {
+
   m.def(
       "is_events",
       [](const DataArrayConstView &self) { return is_events(self); },
       R"(Return true if the data array contains event data. Note that data may be stored as a scalar, but this returns true if any coord contains events.)");
+
+  m.def("less",
+        [](const VariableConstView &x, const VariableConstView &y) {
+          return less(x, y);
+        },
+        py::arg("x"), py::arg("y"),
+        R"(
+        Comparison returning the truth value of (x < y) element-wise.
+
+        :raises: If the units of inputs are not the same, or if the dtypes of inputs are not double precision floats
+        :return: Variable of booleans.
+        :rtype: Variable)");
+
+  m.def("greater",
+        [](const VariableConstView &x, const VariableConstView &y) {
+          return greater(x, y);
+        },
+        py::arg("x"), py::arg("y"),
+        R"(
+        Comparison returning the truth value of (x > y) element-wise.
+
+        :raises: If the units of inputs are not the same, or if the dtypes of inputs are not double precision floats
+        :return: Variable of booleans.
+        :rtype: Variable)");
+
+  m.def("greater_equal",
+        [](const VariableConstView &x, const VariableConstView &y) {
+          return greater_equal(x, y);
+        },
+        py::arg("x"), py::arg("y"),
+        R"(
+        Comparison returning the truth value of (x >= y) element-wise.
+
+        :raises: If the units of inputs are not the same, or if the dtypes of inputs are not double precision floats
+        :return: Variable of booleans.
+        :rtype: Variable)");
+
+  m.def("less_equal",
+        [](const VariableConstView &x, const VariableConstView &y) {
+          return less_equal(x, y);
+        },
+        py::arg("x"), py::arg("y"),
+        R"(
+        Comparison returning the truth value of (x <= y) element-wise.
+
+        :raises: If the units of inputs are not the same, or if the dtypes of inputs are not double precision floats
+        :return: Variable of booleans.
+        :rtype: Variable)");
+
+  m.def("equal",
+        [](const VariableConstView &x, const VariableConstView &y) {
+          return equal(x, y);
+        },
+        py::arg("x"), py::arg("y"),
+        R"(
+        Comparison returning the truth value of (x == y) element-wise.
+
+        :raises: If the units of inputs are not the same, or if the dtypes of inputs are not double precision floats
+        :return: Variable of booleans.
+        :rtype: Variable)");
+
+  m.def("not_equal",
+        [](const VariableConstView &x, const VariableConstView &y) {
+          return not_equal(x, y);
+        },
+        py::arg("x"), py::arg("y"),
+        R"(
+        Comparison returning the truth value of (x != y) element-wise.
+
+        :raises: If the units of inputs are not the same, or if the dtypes of inputs are not double precision floats
+        :return: Variable of booleans.
+        :rtype: Variable)");
+
```

Comment:
Are these exports what we want? I had imagined we might have supported `__lt__, __le__, __gt__, __ge__, __eq__` to give overloads?

Reply:
Not as far as I understood initial story requirements. The idea was to have functions similar to numpy equivalents.
@SimonHeybrock ?

Reply:
Piotr is right. `__eq__` returns true or false and adding the other operators would be inconsistent, in particular since `Variable` also contains a unit. Comparing `Variable` with, e.g., `<` does not make sense, since the result is not true or false, and having anything else as return type is quite non-standard.

Reply:
Ok. That makes sense.